Microsoft.Toolkit.Uwp.SampleApp/SamplePages/CameraHelper/CameraHelperPage.xaml

```diff
@@ -0,0 +1,28 @@
+﻿<Page
+    x:Class="Microsoft.Toolkit.Uwp.SampleApp.SamplePages.CameraHelperPage"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="using:Microsoft.Toolkit.Uwp.SampleApp.SamplePages"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d">
+    <Page.Resources>
+        <Style x:Name="MessageStyle" TargetType="TextBlock">
+            <Setter Property="FontSize" Value="14"/>
+            <Setter Property="FontWeight" Value="Bold"/>
+            <Setter Property="Foreground" Value="Red"/>
+        </Style>
+    </Page.Resources>
+    <StackPanel Orientation="Vertical" Margin="20">
+        <ComboBox x:Name="FrameSourceGroupCombo" Header="Frame Source Group" HorizontalAlignment="Left" Width="Auto">
+            <ComboBox.ItemTemplate>
+                <DataTemplate>
+                    <TextBlock Text="{Binding DisplayName}"></TextBlock>
+                </DataTemplate>
+            </ComboBox.ItemTemplate>
+        </ComboBox>
+        <TextBlock x:Name="CameraErrorTextBlock" Style="{StaticResource MessageStyle}"  Margin="0,0,0,10" Visibility="Collapsed"></TextBlock>
+        <Button x:Name="CaptureButton" Content="Capture Video Frame" Margin="0,10" Click="CaptureButton_Click"></Button>
+        <Image x:Name="CurrentFrameImage" MinWidth="300" Width="400" HorizontalAlignment="Left"></Image>
```

Comment:
instead of using an Image, you can use a MediaPlayerElement XAML control and hook it to your source directly (like in my email) so you don't have to orchestrate rendering of the frames as they come in

Reply:
this would save you most of the code in your page below

Reply:
This is just a different example of using the Video Frames and displaying in an image using the Camera Helper directly. The CameraPreview sample page demonstrates what you mentioned and it uses the xaml control encapsulating the MediaPlayerElement. Does that make sense?

Reply:
yep ok